### PR TITLE
Create board model

### PR DESCRIPTION
**Pull request by psmishenko**
https://trello.com/c/NTh6nwWi/32-introduce-board-model

Снимок экрана от 2022-05-18 22-26-15
Снимок экрана от 2022-05-18 22-26-26
Снимок экрана от 2022-05-18 22-26-33

### CODE REVIEW

File: app/helpers/boards_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module BoardsHelper
```
Comment (B0RlS) [inline app/helpers/boards_helper.rb]: Do you realy need it?

File: app/controllers/boards_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class BoardsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :current_board, only: [:show, :edit, :update, :destroy]
```
Comment (AnastasiyaKireyeva) [inline app/controllers/boards_controller.rb]: you can use except instead

File: app/controllers/boards_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def new
+    @board = Board.new
+  end
+
+  def create
+    board = Board.create(user_id: current_user.id, 
+      name: board_params['name'], 
+      description: board_params['description'])
```
Comment (AnastasiyaKireyeva) [inline app/controllers/boards_controller.rb]:
```suggestion
    board = Board.create!(user_id: current_user.id, 
                                         name: board_params['name'], 
                                         description: board_params['description'])
```

Comment (AnastasiyaKireyeva) [reply app/controllers/boards_controller.rb]: also you can use the same board_params as in update action ( with the help of `merge` method )

**Comment by AnastasiyaKireyeva** [comment]
I think it should should look more attractive, in views I mean

File: app/controllers/boards_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  before_action :authenticate_user!
+  before_action :current_board, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @boards = Board.all
+  end
+
+  def show
+
+  end
```
Comment (AnastasiyaKireyeva) [inline app/controllers/boards_controller.rb]: do not leave empty methods like this

File: app/controllers/boards_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    redirect_to boards_path
+  end
+
+  private
+
+  def board_params
+    params.require(:board).permit(:name, :description)
+  end
+
+  def current_board
```
Comment (AnastasiyaKireyeva) [inline app/controllers/boards_controller.rb]:
```suggestion
  def find_board
```

File: app/controllers/boards_controller.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  def destroy
+    @board.destroy
+
+    redirect_to boards_path
+  end
+
+  private
+
+  def board_params
+    params.require(:board).permit(:name, :description)
```
Comment (AnastasiyaKireyeva) [inline app/controllers/boards_controller.rb]:
```suggestion
    params.require(:board)
          .permit(:name,
                  :description)
```

**Comment by AnastasiyaKireyeva** [comment]
If you delete some file, please delete also it's spec

File: app/views/boards/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    <%= form.label :name, class: "form-label" %><br />
+    <%= form.text_field :name, placeholder: "name", class: "form-control" %>
+  </div>
+   <div class="form-group w-50">
+  <%= form.label :description, class: "form-label" %><br />
+  <%= form.text_field :description, placeholder: "description", class: "form-control" %>
+  </div>
+  <%= form.submit class: "btn btn-primary"%>
+<% end %>
+<%= link_to "Back to all boards", boards_path %>
```
Comment (AnastasiyaKireyeva) [inline app/views/boards/edit.html.erb]: Add new line, please

File: app/views/boards/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    <%= form.label :name, class: "form-label" %><br />
+    <%= form.text_field :name, placeholder: "name", class: "form-control" %>
+  </div>
+   <div class="form-group w-50">
+  <%= form.label :description, class: "form-label" %><br />
+  <%= form.text_field :description, placeholder: "description", class: "form-control" %>
+  </div>
+  <%= form.submit class: "btn btn-primary"%>
+<% end %>
+<%= link_to "Back to all boards", boards_path %>
```
Comment (AnastasiyaKireyeva) [inline app/views/boards/new.html.erb]: the same issue with the new line

File: app/views/boards/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+<h1>Board information</h1>
+<h3>Board name: <%= @board.name %></h2>
+<h3>Board description: "<%= @board.description %>"</h2>
+<%= link_to "Back to all boards", boards_path %>
```
Comment (AnastasiyaKireyeva) [inline app/views/boards/show.html.erb]: and here

File: app/views/shared/_navbar.html.erb [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+      <div class="navbar-nav">
+        <a class="nav-link active" aria-current="page" href="/">Home</a>
+      </div>
+    </div>
+  </div>
+</nav>
```
Comment (AnastasiyaKireyeva) [inline app/views/shared/_navbar.html.erb]: and here

File: public/packs/manifest.json [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
       "js.map": [
-        "/packs/js/application-515ee3ab924edbfd4240.js.map"
+        "/packs/js/application-796d6018c2b1188a549a.js.map"
       ]
     }
-  }
+  },
+  "media/fonts/bootstrap-icons.woff2?08efbba7c53d8c5413793eecb19b20bb": "/packs/media/fonts/bootstrap-icons-f2539a50.woff2",
+  "media/fonts/bootstrap-icons.woff?08efbba7c53d8c5413793eecb19b20bb": "/packs/media/fonts/bootstrap-icons-7ff3ca6f.woff"
 }
```
Comment (AnastasiyaKireyeva) [inline public/packs/manifest.json]: one more missing line

File: app/controllers/boards_controller.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+class BoardsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_board, except: %i[index new create]
+
+  def index
+    @boards = Board.all
+  end
+
+  def show; end
```
Comment (B0RlS) [inline app/controllers/boards_controller.rb]: You can delete this method

File: spec/views/boards/new.html.erb_spec.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (B0RlS) [inline spec/views/boards/new.html.erb_spec.rb]: It seems to me that you can delete this empty files

**Pull request merged**